langchain4j-mongodb-atlas/src/main/java/dev/langchain4j/store/embedding/mongodb/MongoDBEmbeddingStore.java

```diff
@@ -0,0 +1,277 @@
+                .map(ignored -> randomUUID())
+                .collect(toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+        List<Double> queryVector = referenceEmbedding.vectorAsList().stream()
+                .map(Float::doubleValue)
+                .collect(toList());
+        long numCandidates = maxResults * maxResultRatio;
+
+        List<Bson> pipeline = Arrays.asList(
+                vectorSearch(
+                        fieldPath("embedding"),
+                        queryVector,
+                        indexName,
+                        numCandidates,
+                        maxResults,
+                        vectorSearchOptions),
+                project(
+                        fields(
+                                metaVectorSearchScore("score"),
+                                include("embedding", "metadata", "text")
+                        )
+                ),
+                match(
+                        Filters.gte("score", minScore)
+                ));
+
+        try {
+            AggregateIterable<MongoDBMatchedDocument> results = collection.aggregate(pipeline, MongoDBMatchedDocument.class);
+
+            return StreamSupport.stream(results.spliterator(), false)
+                    .map(MappingUtils::toEmbeddingMatch)
+                    .collect(Collectors.toList());
+
+        } catch (MongoCommandException e) {
+            if (log.isErrorEnabled()) {
+                log.error("Error in MongoDBEmbeddingStore.findRelevant", e);
+            }
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void addInternal(String id, Embedding embedding, TextSegment embedded) {
+        addAllInternal(singletonList(id), singletonList(embedding), embedded == null ? null : singletonList(embedded));
+    }
+
+    private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        if (isNullOrEmpty(ids) || isNullOrEmpty(embeddings)) {
+            log.info("do not add empty embeddings to MongoDB Atlas");
+            return;
+        }
+        ensureTrue(ids.size() == embeddings.size(), "ids size is not equal to embeddings size");
+        ensureTrue(embedded == null || embeddings.size() == embedded.size(), "embeddings size is not equal to embedded size");
+
+        List<MongoDBDocument> documents = new ArrayList<>(ids.size());
+        for (int i = 0; i < ids.size(); i++) {
+            MongoDBDocument document = toMongoDbDocument(ids.get(i), embeddings.get(i), embedded == null ? null : embedded.get(i));
+            documents.add(document);
+        }
+
+        InsertManyResult result = collection.insertMany(documents);
+        if (!result.wasAcknowledged() && log.isWarnEnabled()) {
+            log.warn("Add document failed, Document={}", documents);
```

Comment:
should we maybe throw an exception here? what if logger is disabled or user does not look there? can be confusing

Reply:
done